### PR TITLE
Update to angular 15

### PR DESCRIPTION
**Pull request by NicMcPhee**
This basically copies over the latest files from [the Stackblitz project](https://angular.io/generated/live-examples/getting-started-v0/stackblitz.html) and merges those changes into our project.

Other than jumping up to V15 of Angular, most of this is pretty minor. A few of the highlights are:

- They seemed to have removed the environment files altogether (which we were never using)
- They've replaced the use of `polyfills.ts` with `zone.js`
- They've removed some testing bits; I suspect they gave up on adding tests to this, at least for now? 
- They changed the name of the project to `angular.io-example` (no particular idea why)

In making these changes I appear to have removed all support for `ng lint` by mistake. Ooops. Happily, running `ng add @angular-eslint/schematics` totally fixed things locally. Unfortunately our GitHub Actions check still failed.

### CODE REVIEW

**Review by jpwalbran — APPROVED** [review]
Looks good!

**Review by kklamberty — COMMENTED** [review]
I will run this in the lab tomorrow before approving the changes, but everything looks good to me.

File: src/app/app.module.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { ProductListComponent } from './product-list/product-list.component';
     ReactiveFormsModule,
     RouterModule.forRoot([
       { path: '', component: ProductListComponent },
-    ], { useHash: true })
```
Comment (kklamberty) [inline src/app/app.module.ts:16]: Apparently, we need this for GH pages to work

File: src/index.html [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <title>Angular Getting Started</title>
+    <base href="/" />
```
Comment (kklamberty) [inline src/index.html:6]: And, we need to remove this for GH Pages to work.

**Review by kklamberty — COMMENTED** [review]
We broke GH Pages with this commit, but somehow we didn't notice. @floogulinc found what to fix, and I've marked them in this review.

**Pull request merged**